### DISCUSSION TITLE
Adding human oracle

### DISCUSSION THREAD

**Discussion by KBodolai** [baal-org/baal#105] [Q&A]
Hi all, 

I recently discovered ball while looking into active learning, and I'm very excited about it, 

I'm currently trying to use this approach with a set of unlabeled images, where instead of pulling the label out of the original dataset (no labels!), some kind of GUI popped up, so you can actively label the images. I was pretty sure someone would have implemented this already, but I couldn't find anything around. 

Anyone knows if someone has worked on this? if not, I'm happy to give it a go!

K.

**Comment by Dref360** [comment]
Hello,

We do not have a publicly available labelling tool that leverages BaaL sadly.

Did you have a labelling tool in mind that you wanted to integrate with?

I know that [Label Studio](https://labelstud.io/) can support active learning, but they do not support the active learning loop (retrain every N examples, recompute uncertainty), so we would have to do it manually. Other than that, it should be pretty straight-forward.

Hopefully this help!

**Reply by KBodolai** [reply]
Hi, 

I didn't have anything specific in mind, I had a couple of ideas of trying it with basic IPython widgets to pop a carousel with images and an label dropdown below each image (similar to fast.ai's [ImageClassifierCleaner](https://docs.fast.ai/vision.widgets.html)), so it could be done in loop, guess it's time to start exploring!

I already have a hacky version of this in a more manual way, but having access to a smashing tool like BaaL, I really wanted to try to do it 'the right way'.

I just wanted to know if there was some work done down this road, so yes, your answer was very helpful, thanks!

I'll let you know if I come up with something good.

Cheers!
K.

**Reply by Dref360** [reply]
FYI we will add a LabelStudio component in the next release!
https://github.com/ElementAI/baal/discussions/117

**Comment by Dref360** [comment]
Hello!

We release a new blog post on how to use Label Studio:
https://baal.readthedocs.io/en/latest/tutorials/label-studio.html

**Reply by KBodolai** [reply]
Thanks for this, it looks fantastic!

Hopefully I'll get to try it out during the weekend 👍🏼
